### ISSUE TITLE
npm repository error.

### ISSUE THREAD

**Issue by makkaba** [nchutchind/cordova-plugin-streaming-media#72]
**What version of Streaming-Meda-Cordova-Plugin are you using?**
latest version through npm 
**What version of Cordova are you using?**
6.2.0
**What devices are affected?**
npm installing with 'cordova plugin add' command  is failed.
**Please describe the issue in detail, with relevant code samples**
npm installing with 'cordova plugin add' command  is failed.
**What did you expect to happen?**
npm repository url mismatching, probably.
**What actually happened?**
this error message comes out.--------
Discovered plugin "com.hutchind.cordova.plugins.streamingmedia" in config.xml. Adding it to the project
Fetching plugin "com.hutchind.cordova.plugins.streamingmedia@~0.1.4" via npm
Failed to restore plugin "com.hutchind.cordova.plugins.streamingmedia" from config.xml. You might need to try adding it again. Error: Failed to fetch plugin com.hutchind.cordova.plugins.streamingmedia@~0.1.4 via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Error: Registry returned 404 for GET on https://registry.npmjs.org/com.hutchind.cordova.plugins.streamingmedia

---

**Comment by blattmann** [comment]
@makkaba 
Could you solve this issue somehow? I am having the same issue now.

**Comment by raphaelmerx** [comment]
We solved this by committing the `plugins/fetch.json` file to the repository. This file is used by cordova to identify that the `com.hutchind.cordova.plugins.streamingmedia` plugin should be fetched at https://registry.npmjs.org/cordova-plugin-streaming-media, instead of https://registry.npmjs.org/com.hutchind.cordova.plugins.streamingmedia.

If you're not committing this file, then one developer will run `cordova plugin save cordova-plugin-streaming-media`, which will make `com.hutchind.cordova.plugins.streamingmedia` point to the right registry URL in their `fetch.json` file, but other developers will not have that information stored in their `fetch.json`.

In summary, add this to your `fetch.json`, and commit it:
```
    "com.hutchind.cordova.plugins.streamingmedia": {
        "source": {
            "type": "registry",
            "id": "cordova-plugin-streaming-media"
        },
        "is_top_level": true,
        "variables": {}
    }
```

**Comment by blattmann** [comment]
Thanks! I will give it a try!



2017-06-22 8:18 GMT+02:00 Raphael Merx <notifications@github.com>:

**Comment by raphaelmerx** [comment]
@blattmann I deleted my comment because it wasn't working either. Try sticking this line in your `config.xml`:
```
<plugin name="cordova-plugin-streaming-media" spec="1.0.2" />
```
 instead of the line that is added when saving the plugin. It's not ideal as cordova will try to fetch the plugin on every `cordova prepare`, but the request is cached by npm so you don't actually need to have an Internet connection for `cordova prepare` to run.

**Comment by fdambrosio** [comment]
From yesterday I have this error when I run `ionic cordova build ios --release`

```
An error occurred while running cordova prepare (exit code 1):
Discovered plugin "cordova-plugin-streaming-media" in config.xml. Adding it to the project
        Failed to restore plugin "cordova-plugin-streaming-media" from config.xml
```

**Comment by fizshoemaker** [comment]
I'm getting similar error:
`npm ERR! 404 Registry returned 404 for GET on https://registry.npmjs.org/com.hutchind.cordova.plugins.streamingmedia
npm ERR! 404
npm ERR! 404  'com.hutchind.cordova.plugins.streamingmedia' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)`

Any chance this can be fixed? Thanks.

**Comment by fizshoemaker** [comment]
I managed to resolve the issue by changing all reference of "com.hutchind.cordova.plugins.streamingmedia" to "cordova-plugin-streaming-media" in package.json, config.xml and the actual folder name in the plugins folder.

Hope that helps.

**Comment by shamilovtim** [comment]
Resolved with the latest version
